### ISSUE TITLE
HttpClient should not be disposed if provided

### ISSUE THREAD

**Issue by Jericho** [Pathoschild/FluentHttpClient#11]
Say I have the following code in my app:
```
var httpClient = new HttpClient(new HttpClientHandler()
{
    Proxy = new WebProxy(...)
});
using (var fluentClient = new FluentClient("http://example.org/api/", httpClient))
{
    ... do something with fluentClient ...
}
var response = await httpClient.SendAsync(myHttpRequest);
```
This last line will fail because my `httpClient` was disposed by `FluentHttpClient` in the `Dispose(bool isDisposing)` method. If the HtpClient is not passed to the FluentClient constructor, it makes sense for this HttpClient to be disposed. However, FluentClient should never dispose a value that is provided by the developer.

I suggest the following:

1. Add a private field to the FluentClient class: `private readonly bool _mustDisposeHttpClient;`
2. Modify the FluentClient constructors like so:
```
       public FluentClient(string baseUri)
            : this(baseUri, null) { }

        public FluentClient(string baseUri, HttpClient client)
        {
            _mustDisposeHttpClient = client == null;
            this.BaseClient = client ?? new HttpClient();
            this.Filters = new List<IHttpFilter> { new DefaultErrorFilter() };
            if (baseUri != null)
                this.BaseClient.BaseAddress = new Uri(baseUri);
            this.Formatters = new MediaTypeFormatterCollection();
        }
```
3. Finally, modify the `Dispose` method like so:
```
       protected virtual void Dispose(bool isDisposing)
        {
            if (this.IsDisposed)
                return;

            if (isDisposing && _mustDisposeHttpClient)
                this.BaseClient.Dispose();

            this.IsDisposed = true;
        }
```

I will be happy to submit a PR.

**Comment by Pathoschild** [comment]
Makes sense to me. Submit the PR and I'll include it in the 2.3 release.

**Comment by Pathoschild** [comment]
Merged into `develop` for the upcoming 2.4 release. Thanks!
